logging/logger.go

```diff
@@ -0,0 +1,293 @@
+package logging
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+// Logger interface for logging to.
+type Logger interface {
+	ZapCompatibleLogger
+
+	AsZap() *zap.SugaredLogger
+}
+
```

Comment:
```suggestion
var _ Logger = &zLogger{}
```
Or something of the sort?

Reply:
This is cute and clearly describes intent. But I don't think it's necessary/likely to catch anything because our code is consuming Logger/zLogger, this line just adds yet another compile error if `zLogger` no longer satisfies `Logger`
```
logging/logger.go:85:10: cannot use &zLogger{…} (value of type *zLogger) as Logger value in return statement: *zLogger does not implement Logger (missing method AsZap)
logging/logger.go:99:16: cannot use &zLogger{} (value of type *zLogger) as Logger value in variable declaration: *zLogger does not implement Logger (missing method AsZap)
logging/logger.go:107:9: cannot use &zLogger{…} (value of type *zLogger) as Logger value in return statement: *zLogger does not implement Logger (missing method AsZap)
logging/logger.go:230:9: cannot use &zLogger{…} (value of type *zLogger) as Logger value in return statement: *zLogger does not implement Logger (missing method AsZap)
logging/logger.go:239:9: cannot use &zLogger{…} (value of type *zLogger) as Logger value in return statement: *zLogger does not implement Logger (missing method AsZap)
logging/logger.go:248:9: cannot use &zLogger{…} (value of type *zLogger) as Logger value in return statement: *zLogger does not implement Logger (missing method AsZap)
logging/logger.go:257:9: cannot use &zLogger{…} (value of type *zLogger) as Logger value in return statement: *zLogger does not implement Logger (missing method AsZap)
logging/logger.go:273:9: cannot use &zLogger{…} (value of type *zLogger) as Logger value in return statement: *zLogger does not implement Logger (missing method AsZap)
```

Reply:
Ah yeah good point; ignore!